src/quest/persistence.py

```diff
@@ -79,6 +84,73 @@ def has_blob(self, key: str) -> bool:
     def delete_blob(self, key: str):
         self._get_file(key).unlink()
 
+Base = declarative_base()
+
+class RecordModel(Base):
+    __tablename__ = 'records'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    name = Column(String) # TODO good name for this?
+    key = Column(String)
+    blob = Column(JSON)
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__}: {self.name}>'
+
+class SQLDatabase:
+
+    def __init__(self, db_url: str):
+        self._db_url = db_url
+        self._engine = create_engine(db_url)
+
+        Base.metadata.create_all(self._engine)
+
+    def get_engine(self) -> Engine:
+        return self._engine
+
+class SqlBlobStorage(BlobStorage):
```

Comment:
Would exposing BlobStorage to the user to implement allow for easier backend configurability for Quest users? Kind of like serialization? Then we don't have to write a BlobStorage for every type of storage, we just let the user decide where their blobs will be stored?